### PR TITLE
Improve CI setup

### PR DESCRIPTION
**Pull request by nomeata**
Do not fail on pull requests, properly cache dependencies

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -43,16 +43,18 @@ jobs:
...
 
       - name: Build dependencies
         run: stack build --system-ghc --only-dependencies
 
       - name: Build site executable
         run: stack build --system-ghc
 
-      # Runs a set of commands using the runners shell
       - name: Deploy
+        # No not deploy from pull requests
```
Comment (nomeata) [inline .github/workflows/main.yml:56]:
```suggestion
        # No deploy from pull requests
```

**Comment by david-christiansen** [comment]
I am not really competent to review this, TBH. My knowledge of GH actions is limited.

I like that it has a green check mark, and I generally trust @nomeata to say so if his understanding is limited, so I'd be inclined to merge. Anyone else around who has a better understanding than me to check it out?

**Comment by nikivazou** [comment]
I do not have a better understanding, bit I also trust @nomeata and confirm that CI took too long....
So, let's merge :)

**Comment by nomeata** [comment]
We can revert when I messed up (which is possible of course)

**Comment by david-christiansen** [comment]
YOLO :-)

**Comment by nomeata** [comment]
The merge commit pr itself failed because the deploy action doesn't like it if nothing new gets deployed, but that's independent of this PR. Else looks good.

**Pull request merged**